src/RuleSet.js

```diff
@@ -64,6 +64,17 @@ class RuleSet {
         rules.forEach(this.addRule.bind(this));
     }
 
+    /**
+     * Remove a rule from the set. This might happen if a user explicitly sets
+     * a rule to "false" in the set in order to turn off a rule that was turned
+     * on by a different set.
```

Comment:
Generally it feels a little weird to me that one ruleset is allowed to turn off another ruleset's rules. Do we have a real usecase for that? I'd say that enabling that could cause more convoluted definitions compared _enable ruleset A, enable ruleset B, disable rule N_

Reply:
Real use case is this: when there are projects where the file type uses a catch-all ruleset that the plugin publishes, and there is at least one rule in that catch-all ruleset that we don't want. ie. it is "use all the rules that the plugin publishes, EXCEPT these ones."

This comes from trying to run lint on our tm-export xliff files. The tm-export for some reason does not put a "state" attribute on any of the translation units. Also, the "generic" ruleset defined by ilib-lint itself includes the rule that checks that the state is set to "needs review", which fails on every single trans unit. I was stunned to find thousands of errors for each project the first time I ran ilib-lint on our tm-exports until I realized that the vast majority of problems were the missing state attribute. When I was able to turn off the state checker rule, then it became more reasonable and almost all the results were real ones.

As a side note, depressingly, we still have hundreds of errors and warnings per project. We need to finish the autofix stuff framework and implement it for the "array of resources" IR type and various resource rules! That would fix hundreds of problems in our current translations.

Also, we should think about creating a plugin for mojito xliff files. These have a whole bunch of extra information encoded as a json object in a string in the `note` element of each trans unit. This is not standard of course. We could parse that json object and get things like the resource state from that object. Also in there is information about whether the trans unit is part of a plural resource, so we could make an actual PluralResource out of it instead of a StringResource with a fancy comment.

Reply:
What I meant was that a ruleset (either exported by a plugin or custom defined in a config) should only enable rules, whereas disabling should only be possible per-rule. Allowing rulesets to disable rules can get messy since 1. plugin hides ruleset definition so it's not immediately clear that a rule has been disabled 2. if you have 2 rulesets of which one enables and the other disables a given rule, loading order starts to matter.

Reply:
At least this kind of not-so-obvious behaviour should be defined through tests

Reply:
Forgot to push the branch, so it is there now in FileType.test.js

Reply:
How would you suggest disabling a rule if it is not part of a ruleset?